### ISSUE TITLE
Lua Error when submitting crafting order

### ISSUE THREAD

**Issue by icbat** [Auctionator/Journalator#86]
Please include as much of the following information as possible to help me fix the bug:

## Brief Description of Bug

I was submitting a public crafting order. It occurred for both the Centralized Precipitation Emitter and the Environmental Emulator. 

## Steps to Reproduce

1. Visit Clerk Silverpaw in Valdrakken
2. Fill out the Place Order page for a Public Crafting Order
3. Upon pressing submit, receive the LUA Error Output below

## Loaded Addons

Occurs when only running Journalator.

## Back Trace (LUA Error Output)

```lua
Message: bad argument #2 to '?' (Usage: local results = C_CraftingOrders.GetCustomerOptions(params))
Time: Thu Jul 25 21:58:41 2024
Count: 2
Stack: bad argument #2 to '?' (Usage: local results = C_CraftingOrders.GetCustomerOptions(params))
[string "@Interface/AddOns/Journalator/Source/Monitor/CraftingOrderPlacing.lua"]:15: in function <.../Journalator/Source/Monitor/CraftingOrderPlacing.lua:13>
[string "@Interface/AddOns/Journalator/Source/Monitor/CraftingOrderPlacing.lua"]:63: in function <.../Journalator/Source/Monitor/CraftingOrderPlacing.lua:43>
[string "=[C]"]: in function `PlaceNewOrder'
[string "@Interface/AddOns/Blizzard_ProfessionsCustomerOrders/Blizzard_ProfessionsCustomerOrdersForm.lua"]:1580: in function `ListOrder'
[string "@Interface/AddOns/Blizzard_ProfessionsCustomerOrders/Blizzard_ProfessionsCustomerOrdersForm.lua"]:83: in function <...merOrders/Blizzard_ProfessionsCustomerOrdersForm.lua:39>

Locals: skillLineAbilityID = 47454
possibleCraftingOrderOptions = nil
```

**Comment by tomrus88** [comment]
It's because Blizzard added a new non nil'able field to C_CraftingOrders.GetCustomerOptions parameters: currentExpansionOnly

**Comment by mariof81** [comment]
I have the same issue:

21x bad argument #2 to '?' (Usage: local results = C_CraftingOrders.GetCustomerOptions(params))
[string "=[C]"]: in function `GetCustomerOptions'
[string "@Journalator/Source/Monitor/CraftingOrderPlacing.lua"]:15: in function <.../Journalator/Source/Monitor/CraftingOrderPlacing.lua:13>
[string "@Journalator/Source/Monitor/CraftingOrderPlacing.lua"]:63: in function <.../Journalator/Source/Monitor/CraftingOrderPlacing.lua:43>
[string "=[C]"]: in function `PlaceNewOrder'
[string "@Blizzard_ProfessionsCustomerOrders/Blizzard_ProfessionsCustomerOrdersForm.lua"]:1580: in function `ListOrder'
[string "@Blizzard_ProfessionsCustomerOrders/Blizzard_ProfessionsCustomerOrdersForm.lua"]:83: in function <...merOrders/Blizzard_ProfessionsCustomerOrdersForm.lua:39>

Locals:
(*temporary) = "bad argument #2 to '?' (Usage: local results = C_CraftingOrders.GetCustomerOptions(params))"

**Comment by plusmouse** [comment]
Sorry for the delay. This issue is now fixed in 1.90 (available from Wago and Curseforge shortly)
